components/motor/gpio/motor_encoder.go

```diff
@@ -173,8 +177,17 @@ type EncodedMotorState struct {
 	direction    float64
 }
 
-// RPMMonitorStart starts the RPM monitor.
+// rpmMonitorStart starts the RPM monitor.
 func (m *EncodedMotor) rpmMonitorStart() {
+	if m.loop != nil {
+		m.loop.Stop()
+	}
+	if len(m.cfg.ControlLoop.Blocks) != 0 {
```

Comment:
are we re-creating a new control loop each time rpmMonitorStart is called? or is it a one of?

Reply:
yes we are recreating with each every movement call so that the loop can be stopped when the position is reached instead of the motor holding position. I mostly implemented it this way so while I was testing with app it would be less annoying so I'm not opposed to changing it to be constantly running until the motor is closed